### PR TITLE
No error on indirect calls

### PR DESCRIPTION
**Pull request by ajafff**
Fixes: #35866 (in case you want to accept that issue)

@RyanCavanaugh for review

### CODE REVIEW

**Comment by andrewbranch** [comment]
Yes we have

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -27133,8 +27133,9 @@ namespace ts {
                 }
             }
 
-            function isEvalNode(node: Expression) {
-                return node.kind === SyntaxKind.Identifier && (node as Identifier).escapedText === "eval";
+            function isInIndirectCall(node: BinaryExpression) {
+                return node.parent.kind === SyntaxKind.ParenthesizedExpression && (isCallExpression(node.parent.parent) && node.parent.parent.expression === node.parent || node.parent.parent.kind === SyntaxKind.TaggedTemplateExpression) &&
+                    (isAccessExpression(node.right) || isIdentifier(node.right) && node.right.escapedText === "eval");
```
Comment (andrewbranch) [inline src/compiler/checker.ts:27138]: Why do we still have to check for `eval`? I think we want to silence the error for any indirect calls.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Comment by sandersn** [comment]
@ajafff Do you still want to work on this? Otherwise I'd like to close it to reduce the number of open PRs.

**Comment by sandersn** [comment]
Closing since this has gotten stale.

**Pull request closed without merging**